### PR TITLE
104 tms couch and date unique on nights

### LINKED ISSUE #104: Validate couch_id and date composite uniqueness

### PR DESCRIPTION
**Pull request by tmikeschu**
@njgheorghita @Sh1pley @riverswb @ski-climb ready for review, all tests pass.

Add scoped uniqueness validation for night

Date scoped to couch_id must be unique

Closes #104

### CODE REVIEW

File: spec/models/night_spec.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 describe Night do
   describe "validations" do
     it { is_expected.to validate_presence_of(:date) }
+    it { should validate_uniqueness_of(:date).scoped_to(:couch_id) }
```
Comment (ski-climb) [inline spec/models/night_spec.rb:6]: ` it { is_expected.to`...?

File: spec/models/night_spec.rb [hunk 2]
```diff
@@ -15,8 +16,13 @@
     let!(:couch_2) { create(:couch, city: "Newport") }
 
     it "returns only the nights for the given couch" do
-      couch_1.nights << create_list(:night, 5, couch: couch_1)
-      couch_2.nights << create_list(:night, 6, couch: couch_2)
+      5.times do |i|
```
Comment (ski-climb) [inline spec/models/night_spec.rb:19]: Cool setup.  I dig it.

**Review by ski-climb — APPROVED** [review]
Nice, @tmikeschu!  Glad you thought of this - concise implementation too.

File: app/models/night.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 class Night < ApplicationRecord
   validates_presence_of :date
   validate :night_cannot_be_in_the_past
+  validates_uniqueness_of :date, scope: :couch_id
```
Comment (riverswb) [inline app/models/night.rb:4]: Great

**Review by riverswb — APPROVED** [review]
Looks good! Merging

**Pull request merged**